tests/src/test/scala/whisk/core/connector/tests/UserEventTests.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.connector.tests
+
+import akka.actor.ActorSystem
+import common.rest.WskRest
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers
+import whisk.connector.kafka.KafkaConsumerConnector
+import whisk.core.WhiskConfig
+
+import scala.concurrent.duration._
+import common._
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+import whisk.core.connector.{Activation, EventMessage, Metric}
+
+@RunWith(classOf[JUnitRunner])
+class UserEventTests extends FlatSpec with Matchers with WskTestHelpers with StreamLogging {
```

Comment:
Do we need a full blown integration test for this? As it stands, this test is not valid for all deployments (one might turn off user events). How does the test cope for this?

Can we narrow the tests to non-integration tests? i.e. test only the LocalEntitlementProvider if stuff is emitted correctly to a MessageProducer (which you can mock)

Reply:
I see your point, I could add a param in whisk.properties to make the assertions conditionally. Not that beautiful, but good enough to do the job...

Mocking Kafka would is an option, but then we have to loose the test coverage for the invoker path...